androidshared/src/main/java/org/odk/collect/androidshared/ui/SnackbarUtils.kt

```diff
@@ -116,8 +125,23 @@ object SnackbarUtils {
         lastSnackbar?.show()
     }
 
-    data class Action(
+    data class SnackbarDetails @JvmOverloads constructor(
         val text: String,
-        val listener: () -> Unit
+        val action: Action? = null
     )
+
+    data class Action(val text: String, val listener: () -> Unit)
+
+    abstract class SnackbarPresenterObserver<T : Any?>(private val parentView: View) :
```

Comment:
We can probably use this in a few more places, but as with other PRs for v2023.3 I wanted to minimise the merge conflict risk.